### PR TITLE
fixed reset of handles when all rotation axis flags are disabled

### PR DESCRIPTION
**Pull request by thalbern**
- changed reset condition (early out) to allow hiding rotation handles when all axis are disabled 
- made visibility check more readable in rotation handles
- added test condition for rotation handle visibility test to cover bug scenario

## Changes
- Fixes: #8239 

## Verification
added test scenario to existing handle visibility test
verified behavior in bounds control example scene

### CODE REVIEW

File: Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/RotationHandles.cs [hunk 1]
```diff
@@ -294,11 +294,26 @@ private Bounds CreateVisual(int handleIndex, GameObject parent)
...
-                (axisType == CardinalAxisType.Y && config.ShowRotationHandleForY) ||
-                (axisType == CardinalAxisType.Z && config.ShowRotationHandleForZ));
+            if (!IsActive)
+            {
+                return false;
+            }
+            else
+            {
+
+                CardinalAxisType axisType = GetAxisType(handle);
```
Comment (wiwei) [inline Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/RotationHandles.cs:304]:
```suggestion
                CardinalAxisType axisType = GetAxisType(handle);
```

File: Assets/MRTK/Tests/PlayModeTests/Experimental/BoundsControlTests.cs [hunk 2]
```diff
@@ -1042,6 +1042,31 @@ public IEnumerator HandleVisibilityTest()
...
+            boundsControl.gameObject.SetActive(false);
+            yield return null;
+            rotationHandlesConfig.ShowRotationHandleForX = false;
+            rotationHandlesConfig.ShowRotationHandleForY = false;
+            rotationHandlesConfig.ShowRotationHandleForZ = false;
+            boundsControl.BoundsControlActivation = BoundsControlActivationType.ActivateOnStart;
+            boundsControl.gameObject.SetActive(true);
+            yield return null;
+            // refetch transforms
+            rigRoot = boundsControl.transform.Find("rigRoot").gameObject;
```
Comment (CDiaz-MS) [inline Assets/MRTK/Tests/PlayModeTests/Experimental/BoundsControlTests.cs:1056]:
```suggestion
            boundsControl.gameObject.SetActive(true);
            yield return null;
            
            // refetch transforms
            rigRoot = boundsControl.transform.Find("rigRoot").gameObject;
```
nit: spaces for consistency with the rest of the test

File: Assets/MRTK/Tests/PlayModeTests/Experimental/BoundsControlTests.cs [hunk 3]
```diff
@@ -1042,6 +1042,31 @@ public IEnumerator HandleVisibilityTest()
...
+            Assert.IsNotNull(rigRoot, "rigRoot couldn't be found");
+            rotationHandleAxisX = rigRoot.transform.Find("midpoint_0");
+            Assert.IsNotNull(rotationHandleAxisX, "rotation handle couldn't be found");
+            rotationHandleAxisY = rigRoot.transform.Find("midpoint_1");
+            Assert.IsNotNull(rotationHandleAxisY, "rotation handle couldn't be found");
+            rotationHandleAxisZ = rigRoot.transform.Find("midpoint_8");
+            Assert.IsNotNull(rotationHandleAxisZ, "rotation handle couldn't be found");
+            // check handle visibility
+            Assert.IsFalse(rotationHandleAxisX.gameObject.activeSelf, "rotation handle x active");
```
Comment (CDiaz-MS) [inline Assets/MRTK/Tests/PlayModeTests/Experimental/BoundsControlTests.cs:1065]:
```suggestion
            rotationHandleAxisZ = rigRoot.transform.Find("midpoint_8");
            Assert.IsNotNull(rotationHandleAxisZ, "rotation handle couldn't be found");
            
            // check handle visibility
            Assert.IsFalse(rotationHandleAxisX.gameObject.activeSelf, "rotation handle x active");
```

same as above

**Pull request merged**